confidant/routes/credentials.py

```diff
@@ -370,7 +374,9 @@ def update_credential(id):
             }
             return jsonify(ret), 400
         update['credential_pairs'] = json.dumps(credential_pairs)
+        update['last_rotation_date'] = datetime.utcnow()
```

Comment:
There is a key assumption I'm making here that I would like someone to validate  - I'm assuming that if `credential_pairs` is in the payload (`data`), then it has changed.

However, I have looked at the code and I see that it doesn't explicitly check if the key has changed compared to it's previous revisions  - it just checks for conflicts with other keys.  If we think this is a risk, I can fetch all the revisions but I thought that might be overkill..

Reply:
If credential_pairs is in the payload data, it's changed, yeah. That's the only time we should change the rotation time.

Reply:
Though... the client sends in a full object, so I think it's really necessary to compare the values, and only adjust the rotation time if they've changed.

Reply:
>Though... the client sends in a full object, so I think it's really necessary to compare the values, and only adjust the rotation time if they've changed.

What does this mean? It's not referring to  checking that `credential_pairs` is in the payload right?

Reply:
The UI modifies the current object in place, then sends the entire thing in to the update endpoint. That said, it doesn't matter the behavior anyway, because you want to make sure the credentials have actually changed, because they could just send the same value in to fool your rotation detection.